### PR TITLE
KOGITO-7785 Setup init-branch jobs

### PR DESCRIPTION
**Pull request by radtriste**
- https://github.com/kiegroup/kogito-pipelines/pull/591
- https://github.com/kiegroup/drools/pull/4631
- https://github.com/kiegroup/kogito-runtimes/pull/2410
- https://github.com/kiegroup/kogito-apps/pull/1443
- https://github.com/kiegroup/kogito-examples/pull/1380
- https://github.com/kiegroup/kogito-images/pull/1323
- https://github.com/kiegroup/kogito-operator/pull/1264
- https://github.com/kiegroup/kogito-docs/pull/176

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.init-branch [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    FAILED_STAGES.put("${STAGE_NAME}", jobKey)
+}
+void addUnstableStage(String jobKey = '') {
+    UNSTABLE_STAGES.put("${STAGE_NAME}", jobKey)
+}
+
+void sendPipelineErrorNotification() {
+    String bodyMsg = "Kogito init job #${BUILD_NUMBER} was: ${currentBuild.currentResult}"
+
+    paramsStr = ''
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.init-branch]: Another possibility would be
```
    paramsStr = ''
    paramsStr += params.SKIP_TESTS ? '\n- Tests skipped' : ''
    paramsStr += params.SKIP_ARTIFACTS ? '\n- Artifacts skipped' : ''
    paramsStr += params.SKIP_IMAGES ? '\n- Images skipped' : ''
    paramsStr += params.SKIP_OPERATOR ? '\n- Operator skipped' : ''
    bodyMsg += paramsStr ? "\n\nConfiguration:${paramsStr}" : '\n'
```
both are fine, just up to you

File: .ci/jenkins/Jenkinsfile.init-branch [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    }
+    if (params.SKIP_IMAGES) {
+        paramsStr += '\n- Images skipped'
+    }
+    if (params.SKIP_OPERATOR) {
+        paramsStr += '\n- Operator skipped'
+    }
+    bodyMsg += paramsStr ? "\n\nConfiguration:${paramsStr}" : '\n'
+
+    if (FAILED_STAGES.size() > 0) {
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.init-branch]:
```suggestion
    if (FAILED_STAGES.size()) {
```

File: .ci/jenkins/Jenkinsfile.init-branch [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+        bodyMsg += '\nFailed stages: \n- '
+        bodyMsg += FAILED_STAGES.collect { "${it.key} => ${getJobUrl(it.value)}" }.join('\n- ')
+    }
+    bodyMsg += '\n'
+    if (UNSTABLE_STAGES.size() > 0) {
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.init-branch]:
```suggestion
    if (UNSTABLE_STAGES.size()) {
```

**Review by Ginxo — APPROVED** [review]
LGTM, just a couple of minor suggestions

File: .ci/jenkins/Jenkinsfile.init-branch (hunk 1 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.init-branch]: This is a leftover of copy/paste anyway and make no sense in that job. Removed

**Pull request merged**